### PR TITLE
Add conservative surface flux restoring

### PR DESCRIPTION
**Pull request by taimoorsohail**
Surface salinity restoring can carry a nonzero global mean and drift total ocean salt when the model mean differs from the target climatology. This adds `ConservativeSurfaceFluxRestoring`, which materializes the restoring flux and removes its wet-area mean before the boundary condition applies it.

The accompanying callback initializes the corrected field at the start of `run!` and refreshes it once per timestep. The integration test exercises both supported ocean timesteppers and verifies that restoring changes the salinity pattern while conserving volume-integrated salt on CPU and GPU test jobs.

Validation: `git diff --check`. Repository policy delegates test execution to GitHub CI.

### CODE REVIEW

File: src/DataWrangling/restoring.jl [hunk 1]
```diff
@@ -270,6 +270,71 @@ function BoundaryConditions.getbc(sf::SurfaceFluxRestoring, i, j, grid, clock, f
...
+@kernel function _materialize_surface_flux!(buffer, flux, grid, clock, fields)
+    i, j = @index(Global, NTuple)
+    @inbounds buffer[i, j, 1] = BoundaryConditions.getbc(flux, i, j, grid, clock, fields)
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Evaluate `restoring.flux` from the current model state and subtract its
+area-weighted mean over wet cells.
```
Comment (simone-silvestri) [inline src/DataWrangling/restoring.jl:308]:
```suggestion
Evaluate `restoring.flux` from the current model state and subtract its area-weighted mean over wet cells.
```
